### DISCUSSION TITLE
Large number of pages, SSR or ISR?

### DISCUSSION THREAD

**Discussion by kieri097** [vercel/next.js#21834] [Help] [answered]
Hi,

Imagine one would want to recreate let's say MySpace using Next i.e., a potentially very large number of pages. 

I'm trying to figure out whether it would be better to use:

- SSR
Or
- ISR with a low revalidate

My first thought would have been that with SSR you incur a higher computational load but get lower build times. And that with ISR you take the longer build times but need to do less rendering once that is completed. 

However, assuming a fairly active site (hence, a high % of the potential pages will eventually need to be rendered), Next would be rebuilding a significant (and an increasing) number of pages at a high interval. 

The Vercel FAQ on limits says: 
> The maximum number of files that can be uploaded when creating a Deployment is 10,000 for source files and 16,000 for build output files 

My fear is that ISR (again assuming a significant number of pages being built eventually) would need a large amount of disk space/ would lead to a high number of files. Is it possible to store them on an external source, S3 say, and get Next to fetch them from there when required?

Are those actual issues in practice? If so, at what point is ISR not worth it anymore in terms of that trade-off?

Finally, assuming I'd be hosting this on Vercel, does anyone have a view on which of those two options would be more cost-efficient? I'm having a hard time understanding the pricing structure there / what exactly pushing a Next app onto Vercel will spin up in terms of resources....

Many thanks!

**Comment by M1ck0** [comment, marked as answer]
I had a project with about 100k pages. I did the ISR approach. Works like charm.

On build, I had like 100 pages, and others are built only when a user visits them. This way you will ensure that only pages that are active will be built, and you don't want to waste your machine on pages that won't be visited at all.

Everything is hosted on VPS with no issues so far (about 50-60 new pages are created every day).

**Reply by kieri097** [reply]
Interesting, thanks! So that seems to scale pretty effortlessly then... Were you revalidating those pages on a frequent basis as well?

**Comment by M1ck0** [comment]
Yeah, I think I set revalidate to 5 or 10. Not sure exactly but I know that it wasn't bigger than 10.

**Reply by kieri097** [reply]
Fantastic, thanks for your input!

**Reply by M1ck0** [reply]
Glad if I could help :)

**Comment by leerob** [comment]
Small note:

> The maximum number of files that can be uploaded when creating a Deployment is 10,000 for source files and 16,000 for build output files

Note this says _deployment_. Once your site is deployed, you can cache as many pages as you'd like with ISR on Vercel!

**Reply by kieri097** [reply]
Ah perfect, thanks for your reply! I assumed that those would fall under the 16k build output files...

**Comment by Zain-ul-din** [comment]
@leerob @M1ck0 can I use `ISR` and `generateStaticParams` together? and also what about if I have 200k pages and each page takes let say 200 kilo-bytes so what is the storage limits?
